### PR TITLE
[Remove Stream/Optional usage] LPS-174212 Remove_usages_of_Stream_usages_for_module_roles_test

### PR DESCRIPTION
**Pull request by jeffwu0724**
@dantewang , I tried to flip the conditions to buy spaces
FYI, https://github.com/liferay-core-infra/liferay-portal/pull/3238#discussion_r1093004844
Thanks for checking.

### CODE REVIEW

**Comment by jeffwu0724** [comment]
ci:test:roles

File: modules/apps/roles/roles-test/src/testIntegration/java/com/liferay/roles/service/test/RoleLocalServiceTest.java [hunk 1]
```diff
@@ -271,39 +269,30 @@ public void testGetGroupRolesAndTeamRoles() throws Exception {
...
 
-		List<Role> expectedRoles = _roleLocalService.getRoles(companyId);
+		List<Role> expectedRoles = ListUtil.filter(
+			_roleLocalService.getRoles(companyId),
+			role -> {
+				if (excludedRoleNames.contains(role.getName()) ||
+					(role.getType() == RoleConstants.TYPE_ACCOUNT) ||
+					(role.getType() == RoleConstants.TYPE_DEPOT) ||
+					(role.getType() == RoleConstants.TYPE_SITE) ||
+					!role.isTeam()) {
```
Comment (dantewang) [inline modules/apps/roles/roles-test/src/testIntegration/java/com/liferay/roles/service/test/RoleLocalServiceTest.java:279]: @jeffwu0724 What's the reason to merge `role.isTeam()` check into this condition?

Comment (jeffwu0724) [reply modules/apps/roles/roles-test/src/testIntegration/java/com/liferay/roles/service/test/RoleLocalServiceTest.java:279]: Hi @dantewang ,
My first attempt to buy the sapce is return false right after the frist if. But in this way, we do not really reduce the number of lines. 
Therefore, based on my previous change
```
if (!excludedRoleNames.contains(role.getName()) &&
(role.getType() != RoleConstants.TYPE_ACCOUNT) &&
					(role.getType() != RoleConstants.TYPE_DEPOT) &&
					(role.getType() != RoleConstants.TYPE_SITE)) {

					if (role.getType() != RoleConstants.TYPE_PROVIDER) {
						return true;
					}

					if (!role.isTeam()) {
						return false;
					}

					Team team = _teamLocalService.fetchTeam(role.getClassPK());

					if (team == null) {
						return false;
					}

					return team.getGroupId() == groupId;
				}

				return false;
			});
``` 
For both the outer if and if (!role.isTeam()) will return false if we did not match the condition, then I think we can combine them together.

Please correct me if I made any mistakes. Really appreciate your help!

Comment (dantewang) [reply modules/apps/roles/roles-test/src/testIntegration/java/com/liferay/roles/service/test/RoleLocalServiceTest.java:279]: @jeffwu0724 I'm just wondering if the order of the conditions matter. The original logic checks the `role.getType()` first, and then `role.isTeam()` related logic. So, if the type is not `TYPE_PROVIDER`, even if the `role` is not `team`, this should return true. I'd say it's better for us to keep the original order.

Also, "buy space" usually means to reduce the levels of indents, so the lines of code don't need to be shorter after "buying" space.

**Pull request closed without merging**